src/main/java/org/openrewrite/java/testing/cucumber/CucumberAnnotationToSuite.java

```diff
@@ -0,0 +1,74 @@
+package org.openrewrite.java.testing.cucumber;
+
+import java.time.Duration;
+import java.util.List;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.*;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.J.ClassDeclaration;
+
+public class CucumberAnnotationToSuite extends Recipe {
+
+    private static final String IO_CUCUMBER_JUNIT_PLATFORM_ENGINE_CUCUMBER = "io.cucumber.junit.platform.engine.Cucumber";
+    private static final AnnotationMatcher ANNOTATION_MATCHER = new AnnotationMatcher(
+            IO_CUCUMBER_JUNIT_PLATFORM_ENGINE_CUCUMBER);
+
+    private static final String SUITE = "org.junit.platform.suite.api.Suite";
+    private static final String SELECT_CLASSPATH_RESOURCE = "org.junit.platform.suite.api.SelectClasspathResource";
+
+    @Override
+    public String getDisplayName() {
+        return "Replace @Cucumber with @Suite";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace @Cucumber with @Suite and @SelectClasspathResource(\"cucumber/annotated/class/package\").";
+    }
+
+    @Override
+    public @Nullable Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(2);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new UsesType<>(IO_CUCUMBER_JUNIT_PLATFORM_ENGINE_CUCUMBER);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.ClassDeclaration visitClassDeclaration(ClassDeclaration cd, ExecutionContext p) {
+                J.ClassDeclaration classDecl = super.visitClassDeclaration(cd, p);
+                if (classDecl.getAllAnnotations().stream().noneMatch(ANNOTATION_MATCHER::matches)) {
+                    return classDecl;
+                }
+
+                maybeRemoveImport(IO_CUCUMBER_JUNIT_PLATFORM_ENGINE_CUCUMBER);
+                maybeAddImport(SUITE);
+                maybeAddImport(SELECT_CLASSPATH_RESOURCE);
+
+                return classDecl.withLeadingAnnotations(ListUtils.flatMap(classDecl.getLeadingAnnotations(), ann -> {
+                    String code = "@Suite\n@SelectClasspathResource(\"#{}\")";
+                    String path = classDecl.getType().getPackageName().replace('.', '/');
+                    JavaTemplate template = JavaTemplate.builder(this::getCursor, code)
+                            .javaParser(
+                                    () -> JavaParser.fromJavaVersion().classpath("junit-platform-suite-api").build())
```

Comment:
can it be assumed that any project using the `io.cucumber.platform` will also have a `junit-platform-suite-api` dependency?  If not then adding the dependency and using a stub for the template would be a better option

Reply:
In [the latest junit5 example](https://github.com/cucumber/cucumber-jvm/blob/main/examples/calculator-java-junit5/pom.xml#L52) I think we recommend people add both dependencies; is that still correct @mpkorstanje?

Reply:
Didn't want to push any conflicts in case you're working through those double annotation template issues.
Am I correct in thinking the stubs would be something along these lines?
```java
.javaParser(() -> JavaParser.fromJavaVersion().dependsOn(
        "package org.junit.platform.suite.api;\npublic @interface Suite {}",
        "package org.junit.platform.suite.api;\npublic @interface SelectClasspathResource {\nString value();\n}")
    .build())
```
And that would then mean we can remove the build.gradle dependency, while adding `junit-platform-suite-api` as an additional step?

Reply:
The two depencies that are needed `cucumber-junit-platform-engine` and `junit-platform-suite`. Note this isn't the same as the `junit-platform-suite-api`.

The explanation as to why these two depencies is a bit complicated.

Reply:
@timtebeek your JavaParser stubs look correct

Reply:
thanks @mpkorstanje and @timtebeek, I will fix up the JavaTemplate JavaParser and  dependencies, then push the changes :)